source/common/json/json_streamer.h

```diff
@@ -6,20 +6,47 @@
 
 #include "envoy/buffer/buffer.h"
 
+#include "absl/strings/str_join.h"
 #include "absl/strings/string_view.h"
 #include "absl/types/variant.h"
 
 namespace Envoy {
 namespace Json {
 
+class BufferTemplate {
+public:
+  BufferTemplate(Buffer::Instance& buffer) : buffer_(buffer) {}
+
+  void add(absl::string_view a) { buffer_.addFragments({a}); }
+
+  void add(absl::string_view a, absl::string_view b, absl::string_view c) {
+    buffer_.addFragments({a, b, c});
+  }
+
+  Buffer::Instance& buffer_;
+};
+
+class StringTemplate {
+public:
+  StringTemplate(std::string& buffer) : string_(buffer) {}
+
+  void add(absl::string_view a) { absl::StrAppend(&string_, a); }
+  void add(absl::string_view a, absl::string_view b, absl::string_view c) {
+    absl::StrAppend(&string_, a, b, c);
+  }
+
+private:
+  std::string& string_;
+};
```

Comment:
I will think single `addFragments` is good enough for this scenario. By this way, although we cannot take the advantage of the `absl::StrAppend`. But it is fine from my previous benchmark results.
And note, if string is used, I assume it's used for short JSON serialization and we will pre-allocated enough buffer for the string, `absl::StrAppend` make no sense in this case.


```

using StringTemplate = Envoy::Buffer::Instance;

class StringTemplate {
public:
  StringWriter(size_t initial_buffer_size = 2048) { buffer_.reserve(initial_buffer_size); }

  void addFragments(absl::Span<const absl::string_view> fragments) {
    for (absl::string_view fragment : fragments) {
      buffer_.append(fragment.data(), fragment.size());
    }
  }
  std::string buffer_;
};

```